lib/pbench/test/functional/agent/cli/commands/test_config.py

```diff
@@ -0,0 +1,25 @@
+import os
+import pathlib
+import pytest
+
+
+def test_pbench_run(monkeypatch, tmpdir, agent_config, pbench_run):  # noqa F811
+    """Use pbench_run value from pbench-agent.cfg file"""
+    monkeypatch.delenv("pbench_run")
+    env_pbench_run = os.environ.get("pbench_run")
+    command = ["pbench-list-tools"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert exitcode == 0
+    assert env_pbench_run is None
```

Comment:
This is fine as-is, but I would have put the `env_pbench_run` assertion immediately after its assignment:  we want to ensure that the _input_ to the command is an undefined value for `pbench_run`; whereas, with this arrangement what we are checking is that the value is undefined when the command _returns_.  (I suppose, for complete paranoia, we should do the checks in _both_ places....)

Reply:
The input to the command is a `pbench_run` value from `config` file. As the value for environment value is None it seeks the value from the config file and works with that and as it has been able to use the `pbench_run` value from the config file the command is running without any errors.

I must be unable to understand the context here

Reply:
The `assert` is at the end, but it's based on the environment value captured at the *beginning*, so I don't really understand Webb's take on this. Maybe it would be a bit more clear if the `assert` was up front. Then again ... ultimately this really is a test of the `monkeypatch.delenv`; because failure of that call is the only way the `assert` would fail. Generally, we focus on testing our code, not the testing infrastructure, so this is perhaps excessively paranoid.

I'm not sure if we want to test whether `pbench-list-tools` *creates* a `"pbench_run"` environment variable ... or whether there's any reason to care.

Reply:
> The input to the command is a `pbench_run` value from `config` file. As the value for environment value is `None` it seeks the value from the config file and works with that and as it has been able to use the `pbench_run` value from the config file the command is running without any errors.

My apologies, @riya-17.  Let me expand.

The code-under-test checks the environment as well as the configuration file, so, in that sense, there are two inputs which the unit test must control.  That is, this test should ensure that the environment variable is not defined, in order to force the CUT to obtain its value from the configuration file, and then it should assert that the value that the CUT used matches the one in the configuration file.

In total, the unit tests should drive the CUT through all three behaviors -- using the value from the `pbench_run` environment variable, using the value from the configuration file, and using the fall-back value -- and it should set up the pre-conditions for each test so that it can detect whether the CUT is producing the desired result.